substrate/frame/assets/src/functions.rs

```diff
@@ -953,46 +953,59 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	/// Do set metadata
 	pub(super) fn do_set_metadata(
 		id: T::AssetId,
-		from: &T::AccountId,
 		name: Vec<u8>,
 		symbol: Vec<u8>,
 		decimals: u8,
+		request: MetadataRequestOf<T::AccountId>,
 	) -> DispatchResult {
 		let bounded_name: BoundedVec<u8, T::StringLimit> =
 			name.clone().try_into().map_err(|_| Error::<T, I>::BadMetadata)?;
 		let bounded_symbol: BoundedVec<u8, T::StringLimit> =
 			symbol.clone().try_into().map_err(|_| Error::<T, I>::BadMetadata)?;
 
+		let (is_frozen, is_force_origin) = request.origin();
 		let d = Asset::<T, I>::get(&id).ok_or(Error::<T, I>::Unknown)?;
 		ensure!(d.status == AssetStatus::Live, Error::<T, I>::AssetNotLive);
-		ensure!(from == &d.owner, Error::<T, I>::NoPermission);
+		if !is_force_origin {
+			ensure!(request.account::<T, I>()? == &d.owner, Error::<T, I>::NoPermission);
+		}
 
 		Metadata::<T, I>::try_mutate_exists(id.clone(), |metadata| {
-			ensure!(metadata.as_ref().map_or(true, |m| !m.is_frozen), Error::<T, I>::NoPermission);
+			ensure!(
+				is_force_origin || metadata.as_ref().map_or(true, |m| !m.is_frozen),
+				Error::<T, I>::NoPermission
+			);
 
 			let old_deposit = metadata.take().map_or(Zero::zero(), |m| m.deposit);
-			let new_deposit = Self::calc_metadata_deposit(&name, &symbol);
 
-			if new_deposit > old_deposit {
-				T::Currency::reserve(from, new_deposit - old_deposit)?;
+			let deposit = if is_force_origin {
+				old_deposit
```

Comment:
Why?

Reply:
That was just to reproduce the implementation already in place. If the call is made from a force origin, it used the old deposit.

Reply:
OK fair enough. I think with the force origin behind governance, giving it no deposit is sensible.